### PR TITLE
added info to list resources in repo

### PR DESCRIPTION
**Pull request by andypetrella**
@maasg this PR adds (principally) the `lists` function to load from the repo

### CODE REVIEW

**Comment by maasg** [comment]
test don't pass. fixing...

> [info] *** 1 SUITE ABORTED ***
> [error] Error during tests:
> [error] 	notebook.io.FileSystemNotebookProviderTests
> [error] (test:test) sbt.TestsFailedException: Tests unsuccessful
>

File: src/main/scala/notebook/io/NotebookProvider.scala [hunk 1]
```diff
@@ -1,25 +1,58 @@
...
+            n, dropRoot(f)
+          )
+        }
+      }
+    }
+  }
 }
 
 class FileSystemNotebooksProvider extends NotebookProvider {
+  var config:Config = null //aouch
```
Comment (radekg) [inline src/main/scala/notebook/io/NotebookProvider.scala]: `private`?

File: src/main/scala/notebook/NBSerializer.scala [hunk 2]
```diff
@@ -216,6 +216,15 @@ object NBSerializer {
 
   implicit val worksheetFormat = Json.format[Worksheet]
 
+  sealed trait Resource {
```
Comment (radekg) [inline src/main/scala/notebook/NBSerializer.scala]: The `sealed` trait will bite you hard in the future.

File: src/main/scala/notebook/NBSerializer.scala [hunk 3]
```diff
@@ -216,6 +216,15 @@ object NBSerializer {
...
+    def name:String
+    def path:String
+  }
+
+  case class GenericFile(name:String, path:String, tpe:String) extends Resource
```
Comment (radekg) [inline src/main/scala/notebook/NBSerializer.scala]: I'd add `val`s everywhere.

File: src/main/scala/notebook/io/NotebookProvider.scala (hunk 1 above)
Comment (radekg) [inline src/main/scala/notebook/io/NotebookProvider.scala]: Why not init to `ConfigFactory.empty()`?

File: src/main/scala/notebook/io/NotebookProvider.scala [hunk 4]
```diff
@@ -1,25 +1,58 @@
...
+
+  override val root = Paths.get(config.getString("notebooks.dir"))
```
Comment (radekg) [inline src/main/scala/notebook/io/NotebookProvider.scala]: `lazy`?

**Pull request merged**